compiler/rustc_middle/src/ty/context.rs

```diff
@@ -220,6 +220,13 @@ impl<'tcx> Interner for TyCtxt<'tcx> {
         self.check_and_mk_args(def_id, args)
     }
 
+    fn mk_canonical_goal_evaluation_step(
```

Comment:
```suggestion
    fn intern_canonical_goal_evaluation_step(
```
I feel like `mk` should only be for `XKind -> X`, so this doesn't feel right to me

Reply:
We use `mk_` either for `Kind` interning and for some `List`s, but I'd be happy to keep it to just those.